deployment-repository/src/main/java/org/jboss/as/repository/ContentRepository.java

```diff
@@ -376,7 +375,9 @@ public void removeContent(ContentReference reference) {
                 }
                 Path grandParent = parent.getParent();
                 try {
-                    Files.deleteIfExists(grandParent);
+                    if (!Files.list(grandParent).findAny().isPresent()) {
+                        Files.deleteIfExists(grandParent);
+                    }
                 } catch (IOException ex) {
```

Comment:
Files.list(Path) notes that the returned stream is a DirectoryStream that should be handled using try-with-resources. So perhaps:

```
try (Stream s = Files.list(grandparent)) {
    if (!s.findAny().isPresent()) {
        Files.deleteIfExists(grandParent);     
    }
}
```